### PR TITLE
 #177 Add organizations user events

### PR DESCRIPTION
**Pull request by rtufisi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ The extensions herein are used in the [Phase Two](https://phasetwo.io) cloud off
     - [Resources](#resources)
     - [Mappers](#mappers)
     - [Admin Events](#admin-events)
+    - [User Events](#user-events)
```
Comment (xgp) [inline README.md:29]: This is getting big for the README. Can you create a page `docs/events.md` and put both the admin and user events in that page? Then you can link from the section below.

File: src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java [hunk 2]
```diff
@@ -16,6 +13,10 @@
 import org.keycloak.models.RealmModel;
 import org.keycloak.provider.ProviderEvent;
 
+import java.util.Map;
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java:16]: Need to run `mvn fmt:format` before commit to keep imports consistent

File: src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java [hunk 3]
```diff
@@ -61,6 +62,10 @@ private void addUser(AuthenticationFlowContext context) {
             org.getName(), context.getUser().getUsername());
         org.grantMembership(context.getUser());
         // TODO default roles from config??
+        context.getEvent()
+                .user(context.getUser())
+                .detail("joinedOrganization", org.getId())
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java:67]: Keys are `snake_case` in details. Change to `joined_organization`

File: src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java [hunk 4]
```diff
@@ -87,11 +93,21 @@ public void processAction(RequiredActionContext context) {
                 // add membership
                 log.infof("selected %s", i.getOrganization().getId());
                 memberFromInvitation(i, user);
-                // todo future tell the inviter they accepted
+                event.clone()
+                        .event(CUSTOM_REQUIRED_ACTION)
+                        .user(user)
+                        .detail("orgId", i.getOrganization().getId())
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java:99]: snake_case

File: src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java [hunk 5]
```diff
@@ -87,11 +93,21 @@ public void processAction(RequiredActionContext context) {
...
+                        .detail("invitationId", i.getId())
+                        .success();
               }
               // revoke invitation
               i.getOrganization().revokeInvitation(i.getId());
-              // todo future tell the inviter they rejected
+              event.clone()
+                      .event(CUSTOM_REQUIRED_ACTION)
+                      .detail("orgId", i.getOrganization().getId())
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java:107]: snake_case

File: src/main/java/io/phasetwo/service/resource/MembersResource.java [hunk 6]
```diff
@@ -69,6 +71,13 @@ public Response removeMember(@PathParam("userId") String userId) {
       if (activeOrganizationUtil.isValid() &&
           activeOrganizationUtil.getActiveOrganization().getId().equals(organization.getId())) {
         member.removeAttribute(ACTIVE_ORGANIZATION);
+
+        EventBuilder event = new EventBuilder(realm, session, connection);
+        event
+                .event(UPDATE_PROFILE)
+                .user(user)
+                .detail("removedActiveOrganizationId", activeOrganizationUtil.getActiveOrganization().getId())
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/resource/MembersResource.java:79]: snake_case

File: src/main/java/io/phasetwo/service/resource/UserResource.java [hunk 7]
```diff
@@ -92,9 +94,17 @@ public Response switchActiveOrganization(
...
     // attribute based active organization
+    var currentActiveOrganization = user.getFirstAttribute(ACTIVE_ORGANIZATION);
     user.setAttribute(ACTIVE_ORGANIZATION, Collections.singletonList(body.getId()));
     TokenManager tokenManager = new TokenManager(session, auth.getToken(), realm, user);
+    EventBuilder event = new EventBuilder(realm, session, connection);
 
+    event
+            .event(UPDATE_PROFILE)
+            .user(user)
+            .detail("newActiveOrganizationId", body.getId())
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/resource/UserResource.java:105]: snake_case

**Review by xgp — CHANGES_REQUESTED** [review]
Looks good. A few comments about imports and using snake_case for event details.

File: src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java (hunk 3 above)
Comment (rtufisi) [reply src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java]: Done

File: src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java (hunk 4 above)
Comment (rtufisi) [reply src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java]: Done

File: src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java (hunk 5 above)
Comment (rtufisi) [reply src/main/java/io/phasetwo/service/auth/invitation/InvitationRequiredAction.java]: Done

File: src/main/java/io/phasetwo/service/resource/MembersResource.java (hunk 6 above)
Comment (rtufisi) [reply src/main/java/io/phasetwo/service/resource/MembersResource.java]: Done

**Review by xgp — APPROVED** [review]

**Pull request merged**